packages/profile-sync-controller/src/controllers/user-storage/services.ts

```diff
@@ -77,6 +94,57 @@ export async function getUserStorage(
   }
 }
 
+/**
+ * User Storage Service - Get all storage entries for a specific feature.
+ *
+ * @param opts - User Storage Options
+ * @returns The storage entry, or null if fails to find entry
+ */
+export async function getUserStorageAllFeatureEntries(
+  opts: UserStorageAllFeatureEntriesOptions,
+): Promise<string[] | null> {
+  try {
+    const { bearerToken, path } = opts;
+    const url = new URL(`${USER_STORAGE_ENDPOINT}/${path}`);
+
+    const userStorageResponse = await fetch(url.toString(), {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    });
+
+    // Acceptable error - since indicates feature does not exist.
+    if (userStorageResponse.status === 404) {
+      return null;
+    }
+
+    if (userStorageResponse.status !== 200) {
+      throw new Error('Unable to get User Storage');
+    }
+
+    const userStorage: GetUserStorageAllFeatureEntriesResponse | null =
+      await userStorageResponse.json();
+
+    if (!Array.isArray(userStorage)) {
+      return null;
+    }
+
+    const decryptedData = userStorage?.flatMap((entry) => {
+      if (!entry.Data) {
+        return [];
+      }
+
+      return encryption.decryptString(entry.Data, opts.storageKey);
```

Comment:
Note - this is good.

Heads up, depending on mobile investigations, encryption and decryption may become async (meaning that we'll may need to await these)